### PR TITLE
Lint/update eslintrc

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Take two on updating eslint.  Fixes #1541 
### Proposed Changes

Updates to eslint 4.16.0
Adds more explicit indentation rules.
Sets indentation to "warn" until I fix the remaining 50 warnings.
Fixes all new eslint errors.

### Reason for Changes

Fix #1541 

### Test Coverage
I ran eslint locally and have 0 errors and 50 warnings.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,6 @@
     "webdriverio": "^4.6.2"
   },
   "devDependencies": {
-    "eslint": "2.9.0"
+    "eslint": "4.16.0"
```
Comment (AnmAtAnm) [inline package.json]: "^4.16.0" for compatible version. See [here](https://docs.npmjs.com/files/package.json#dependencies).

Comment (rachel-fenichel) [reply package.json]: Done.

I knew about the conflict, but wanted to get a pull request open so I could confirm that travis was working.

File: core/workspace_svg.js [hunk 2]
```diff
@@ -388,7 +388,7 @@ Blockly.WorkspaceSvg.prototype.createDom = function(opt_backgroundClass) {
     bottom = this.addTrashcan_(bottom);
   }
   if (this.options.zoomOptions && this.options.zoomOptions.controls) {
-    bottom = this.addZoomControls_(bottom);
+    this.addZoomControls_(bottom);
```
Comment (picklesrus) [inline core/workspace_svg.js]: why did this change?

Comment (rachel-fenichel) [reply core/workspace_svg.js]: The `no-unused-variables` triggered on this: "Bottom is assigned a value but never used."

These functions take in a bottom position and return the top of the object after it has been laid out.  That can then be used to position another icon above it.  In this case, we're not positioning any other icons above.

Comment (picklesrus) [reply core/workspace_svg.js]: Ah, I see. I was thrown by the assignment on 388.  But makes sense.

**Pull request merged**